pkg/netpol/connlist/connlist.go

```diff
@@ -398,23 +419,31 @@ func (ca *ConnlistAnalyzer) getConnectionsList(pe *eval.PolicyEngine, ia *ingres
 		ca.logger.Warnf("connectivity analysis found no allowed connectivity between pairs from the configured workloads or external IP-blocks")
 	}
 
-	return connsRes, peersRes, nil
+	return connsRes, peers, nil
 }
 
-// getConnectionsBetweenPeers returns connections list from PolicyEngine object
-func (ca *ConnlistAnalyzer) getConnectionsBetweenPeers(pe *eval.PolicyEngine) ([]Peer2PeerConnection, []Peer, error) {
-	// get workload peers and ip blocks
-	peerList, err := pe.GetPeersList()
-	if err != nil {
-		ca.errors = append(ca.errors, newResourceEvaluationError(err))
-		return nil, nil, err
+// existsFocusWorkload checks if the provided focus workload is ingress-controller
+// or if it exists in the peers list from the parsed resources
+func (ca *ConnlistAnalyzer) existsFocusWorkload(peerList []eval.Peer) bool {
+	// if focus workload is ingress controller, it is okay to continue checking for connections
+	ingressPodNsNameFormat := types.NamespacedName{Namespace: ingressanalyzer.IngressPodNamespace,
+		Name: ingressanalyzer.IngressPodName}.String()
+	if ca.focusWorkload == ingressanalyzer.IngressPodName || ca.focusWorkload == ingressPodNsNameFormat {
+		return true
```

Comment:
what if focus-workload is ingress-controller, but no Route/Ingress resource exist? (doesn't that mean that ingress-controller is not added at all?)

Reply:
handled